velox/functions/prestosql/types/IPPrefixType.cpp

```diff
@@ -51,8 +80,114 @@ class IPPrefixCastOperator : public exec::CastOperator {
       const TypePtr& resultType,
       VectorPtr& result) const override {
     context.ensureWritable(rows, resultType, result);
-    VELOX_NYI(
-        "Cast from IPPrefix to {} not yet supported", resultType->toString());
+
+    if (resultType->kind() == TypeKind::VARCHAR) {
+      castToString(input, context, rows, *result);
+    } else {
+      VELOX_NYI(
+          "Cast from IPPrefix to {} not yet supported", resultType->toString());
+    }
+  }
+
+ private:
+  static void castToString(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      BaseVector& result) {
+    auto* flatResult = result.as<FlatVector<StringView>>();
+    const auto* ipprefixes = input.as<RowVector>();
+    const auto* ip = ipprefixes->childAt(0)->as<SimpleVector<int128_t>>();
+    const auto* prefix = ipprefixes->childAt(1)->as<SimpleVector<int8_t>>();
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      const auto intAddr = ip->valueAt(row);
+      folly::ByteArray16 addrBytes;
+
+      memcpy(&addrBytes, &intAddr, ipaddress::kIPAddressBytes);
```

Comment:
Can we add a compile time check here that the `sizeof(addrBytes)` and `ipaddress::kIPAddressBytes` are the same?